myapp/spec/system/tasks_spec.rb

```diff
@@ -47,66 +48,124 @@
 
       expect(page).to have_field('Title')
       expect(page).to have_field('Description')
+      expect(page).to have_field('Due')
 
-      expect(page).to have_selector("input[type='submit'][value='Save Task']")
+      expect(page).to have_selector("input[type='submit'][value='Submit Task']")
     end
 
     it 'create new task and show flash message' do 
       visit new_task_path
 
       fill_in 'Title', with: 'New Task Title'
       fill_in 'Description', with: 'New Task Description'
+      new_time = Time.zone.now + 5
+      fill_in 'Due', with: new_time
 
-      click_on 'Save Task'
+      click_on 'Submit Task'
 
       expect(current_path).to eq(tasks_path)
       expect(page).to have_content('New task was created successfully!')
-      expect(Task.all.length).to eq(3)
+      expect(Task.all.length).to eq(1)
 
       visit task_path(Task.last)
 
       expect(page).to have_content('New Task Title')
       expect(page).to have_content('New Task Description')
+      expect(page).to have_content(I18n.l new_time)
+    end
+
+    context 'input invalid value' do 
+      it 'should show error message for empty title' do 
+        visit new_task_path 
+
+        fill_in 'Description', with: 'New Task Description'
+        fill_in 'Due', with: Time.zone.now + 5
+
+        click_on 'Submit Task'
+
+        expect(page).to have_content('You have some invalid inputs!')
+        expect(page).to have_content('Title must not be blank!')
+      end
+
+      it 'should show error message for empty description' do 
```

Comment:
Also need to add test about text length?

Reply:
Added tests for length and removed tests for empty input because the validatioins were deleted.
see 3f5c477